diagrams/architecture.dot

```diff
@@ -0,0 +1,17 @@
+digraph Architecture {
```

Comment:
mermaid diagram would get rendered then no need for the png

Reply:
ive added Mermaid diagram to the readme. it is convenient, but may not look as good